src/StandaloneHydratorPluginManager.php

```diff
@@ -0,0 +1,138 @@
+    ContainerInterface,
+    HydratorPluginManagerInterface
+{
+    /**
+     * To allow using the short name (class name without namespace), this maps
+     * the lowercase name to the FQCN. For hydrators that in previous versions
+     * did not have the Hydrator suffix, it also maps the class name without
+     * the suffix.
+     *
+     * @var array<string, string>
+     */
+    private $aliases = [
+        'arrayserializable'         => ArraySerializableHydrator::class,
+        ArraySerializable::class    => ArraySerializableHydrator::class,
+        'arrayserializablehydrator' => ArraySerializableHydrator::class,
+        ClassMethods::class         => ClassMethodsHydrator::class,
+        'classmethods'              => ClassMethodsHydrator::class,
+        'classmethodshydrator'      => ClassMethodsHydrator::class,
+        'delegatinghydrator'        => DelegatingHydrator::class,
+        ObjectProperty::class       => ObjectPropertyHydrator::class,
+        'objectpropertyhydrator'    => ObjectPropertyHydrator::class,
+        'objectproperty'            => ObjectPropertyHydrator::class,
+        Reflection::class           => ReflectionHydrator::class,
+        'reflectionhydrator'        => ReflectionHydrator::class,
+        'reflection'                => ReflectionHydrator::class,
+    ];
+
+    /**
+     * @var array<string, callable>
+     */
+    private $factories = [];
+
+    /**
+     * @var callable Invokable factory for hydrators without dedicated factories.
+     */
+    private $invokableFactory;
+
+    public function __construct()
+    {
+        $this->invokableFactory = function (ContainerInterface $container, string $class) {
+            return new $class();
+        };
+
+        $this->factories = [
+            ArraySerializableHydrator::class => $this->invokableFactory,
+            ClassMethodsHydrator::class      => $this->invokableFactory,
+            DelegatingHydrator::class        => new DelegatingHydratorFactory(),
+            ObjectPropertyHydrator::class    => $this->invokableFactory,
+            ReflectionHydrator::class        => $this->invokableFactory,
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get($id)
+    {
+        $class = $this->resolveName($id);
+        if (! $class) {
+            throw Exception\MissingHydratorServiceException::forService($id);
+        }
+
+        $instance = ($this->factories[$class])($this, $class);
+
+        return $instance;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function has($id)
+    {
+        return null !== $this->resolveName($id);
+    }
+
+    /**
+     * Resolve a service name from an identifier.
+     *
+     * If $name is registered in $factories, the method returns it verbatim.
+     *
+     * Next it checks if the $name is registered verbatim in $aliases; if so,
+     * it returns the target of the alias.
+     *
+     * finally, it does a strtolower() on it and looks to see if it exists
```

Comment:
This is a description of the implementation: since we are in `private` scope, most of the docblock, whilst extensive and descriptive, is kinda not needed